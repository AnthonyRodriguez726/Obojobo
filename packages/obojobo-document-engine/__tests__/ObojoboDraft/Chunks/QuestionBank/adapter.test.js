import QuestionBankAdapter from '../../../../ObojoboDraft/Chunks/QuestionBank/adapter'

describe('QuestionBank adapter', () => {
	test('construct builds without attributes', () => {
		let model = { modelState: {} }
		QuestionBankAdapter.construct(model)
		expect(model).toMatchSnapshot()
	})

	test('construct builds with attributes', () => {
		let model = { modelState: {} }
		let attrs = {
			content: {
				choose: 2,
				select: 'random-all',
				shuffleGroup: true,
				groupSize: 2
			}
		}

		QuestionBankAdapter.construct(model, attrs)
		expect(model).toMatchSnapshot()
	})

	test('clone creates a copy', () => {
		let a = { modelState: {} }
		let b = { modelState: {} }
		let attrs = {
			content: {
				choose: 2,
				select: 'random-all',
				shuffleGroup: true,
				groupSize: 2
			}
		}

		QuestionBankAdapter.construct(a, attrs)
		QuestionBankAdapter.clone(a, b)

		expect(a).not.toBe(b)
		expect(a).toEqual(b)
	})

	test('toJSON builds a JSON representation', () => {
		let model = { modelState: {} }
		let json = { content: {} }
		let attrs = {
			content: {
				choose: 2,
				select: 'random-all',
				shuffleGroup: true,
				groupSize: 2
			}
		}

		QuestionBankAdapter.construct(model, attrs)
		QuestionBankAdapter.toJSON(model, json)

		expect(json).toMatchSnapshot()
	})
})
