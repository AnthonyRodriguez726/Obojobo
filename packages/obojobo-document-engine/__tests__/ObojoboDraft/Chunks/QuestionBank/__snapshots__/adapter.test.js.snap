// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionBank adapter can be constructed WITH attributes 1`] = `
Object {
  "modelState": Object {
    "choose": 2,
    "groupSize": 2,
    "select": "random-all",
    "shuffleGroup": true,
  },
}
`;

exports[`QuestionBank adapter can be constructed WITHOUT attributes 1`] = `
Object {
  "modelState": Object {
    "choose": Infinity,
    "groupSize": 1,
    "select": "sequential",
    "shuffleGroup": false,
  },
}
`;

exports[`QuestionBank adapter can be converted to JSON 1`] = `
Object {
  "content": Object {
    "choose": 2,
    "groupSize": 2,
    "select": "random-all",
  },
}
`;
